Source/WebCore/svg/graphics/SVGImage.cpp

```diff
@@ -551,6 +551,18 @@ void SVGImage::subresourcesAreFinished(CompletionHandler<void()>&& completionHan
     internalPage()->localTopDocument()->whenWindowLoadEventOrDestroyed(WTFMove(completionHandler));
 }
 
+void SVGImage::setShouldDelayLoadEvent(Document& document, bool shouldDelay)
+{
+    if (m_shouldDelayLoadEvent == shouldDelay)
+        return;
+
+    m_shouldDelayLoadEvent = shouldDelay;
```

Comment:
It is a bit odd to have this method here, given SVGImage is not able to recompute its document.
I would tend to have the bool in ImageLoader if we need one.

Reply:
That would not help with the case I called out in the commit message. Where you have multiple `ImageLoader` instances pointing to the same `SVGImage` instance. Which happens in the various `svg/as-image` tests as they use `svg:use`. But if we merge this into `subresourcesAreFinished()` perhaps the concern goes away?